internal/pipeline/component/pipeline_test.go

```diff
@@ -936,6 +939,740 @@ spec:
           - name: file-mount-d08babc2
             configMap:
               name: test-component-dev-12345678-config-app-properties-7a40d758
+`,
+			wantErr: false,
+		},
+		{
+			name: "component with env and file secrets",
+			snapshotYAML: `
+apiVersion: core.choreo.dev/v1alpha1
```

Comment:
Shall we use testdata directory similar to https://github.com/openchoreo/openchoreo/tree/main/internal/pipeline/component/testdata?